Darktrace/helpers.py

```diff
@@ -0,0 +1,20 @@
+import hmac
+import hashlib
+import requests
+from urllib.parse import urlparse
+
+
+def generate_darktrace_signature(public_key: str, private_key: str, query: str, now: str) -> str:
+    maccer = hmac.new(
+        private_key.encode("ASCII"), (query + "\n" + public_key + "\n" + now).encode("ASCII"), hashlib.sha1
+    )
+    sig = maccer.hexdigest()
+
+    return sig
+
+
+def extract_query(request: requests.models.PreparedRequest) -> str:
+    parsed_url = urlparse(request.url)
+    query = str(parsed_url.path) + "?" + str(parsed_url.query)
```

Comment:
I do not know why mypy didn't like those without str casting. 
Error was :
> error: Unsupported operand types for + ("bytes" and "str")  [operator]
When I checked with type(), they were already strings.

Reply:
The result of `urlparse` is a namedtuple. I wonder if mypy is unable to infer the type of the part of the namedtuple